src/zone.rs

```diff
@@ -175,6 +179,84 @@ impl Zone {
             _ => false,
         }
     }
+
+    pub fn iter_parents<'a>(&'a self, all_zones: &'a MutableSlice) -> ParentIterator<'a> {
+        ParentIterator {
+            zone: &self,
+            all_zones: all_zones,
+        }
+    }
+
+    /// compute a nice human readable label
+    /// The label carries the hierarchy of a zone.
+    ///
+    /// This label is inspired from
+    /// [opencage formatting](https://blog.opencagedata.com/post/99059889253/good-looking-addresses-solving-the-berlin-berlin)
+    ///
+    /// and from the [mimirsbrunn](https://github.com/CanalTP/mimirsbrunn) zip code formatting
+    ///
+    /// example of zone's label:
+    /// Paris (75000-75116), Île-de-France, France
+    pub fn compute_label(&mut self, all_zones: &MutableSlice) {
+        let mut label = format!(
+            "{n}{zip}",
+            n = self.name.clone(),
+            zip = format_zip_code(&self.zip_codes)
+        );
+        {
+            let parent_labels = self.iter_parents(all_zones)
+                .map(|z| z.name.clone())
+                .filter(|n| n != &self.name)
```

Comment:
I'm not fond of this implementation: potentially buggy eg if a country and a city have the same name (Luxembourg...)